pulsar-client/src/main/java/org/apache/pulsar/client/impl/Backoff.java

```diff
@@ -22,29 +22,59 @@
 import java.time.Clock;
 import java.util.Random;
 import java.util.concurrent.TimeUnit;
-import lombok.Data;
+import lombok.Getter;
 
-// All variables are in TimeUnit millis by default
-@Data
```

Comment:
I guess @Data should be replaced with `@ToString @EqualsAndHashCode` if the intention is to remove setters. @Value would be the correct solution if Backoff would be fully immutable.

Reply:
https://projectlombok.org/features/Data is the docs for `@Data`

Reply:
Great point. Do you think that I also need to make the mutable fields `volatile` then, since `toString()` and `.equals` won't be synchronized?

Reply:
> Great point. Do you think that I also need to make the mutable fields `volatile` then, since `toString()` and `.equals` won't be synchronized?

I think that it would be an overkill to use volatile in this case. In Java with 64 bit JVMs you cannot hit a "word tearing" issue where the observed field value would be corrupted. The worst that could happen in Java (on a 64 bit JVM) is a data race.

Reply:
I added in the annotations.

Reply:
Interestingly, it looks like spot bugs didn't like the change for the lack of synchronization introduced by the annotations:

> Error: Medium: Inconsistent synchronization of org.apache.pulsar.client.impl.Backoff.next; locked 75% of time [org.apache.pulsar.client.impl.Backoff, org.apache.pulsar.client.impl.Backoff, org.apache.pulsar.client.impl.Backoff, org.apache.pulsar.client.impl.Backoff, org.apache.pulsar.client.impl.Backoff, org.apache.pulsar.client.impl.Backoff, org.apache.pulsar.client.impl.Backoff, org.apache.pulsar.client.impl.Backoff, org.apache.pulsar.client.impl.Backoff, org.apache.pulsar.client.impl.Backoff, org.apache.pulsar.client.impl.Backoff, org.apache.pulsar.client.impl.Backoff, org.apache.pulsar.client.impl.Backoff] Unsynchronized access at Backoff.java:[line 35]Unsynchronized access at Backoff.java:[line 35]Unsynchronized access at Backoff.java:[line 35]Synchronized access at Backoff.java:[line 96]Synchronized access at Backoff.java:[line 99]Synchronized access at Backoff.java:[line 97]Synchronized access at Backoff.java:[line 101]Synchronized access at Backoff.java:[line 101]Synchronized access at Backoff.java:[line 142]Synchronized access at Backoff.java:[line 133]Synchronized access at Backoff.java:[line 134]Synchronized access at Backoff.java:[line 134]Synchronized access at Backoff.java:[line 34] IS2_INCONSISTENT_SYNC

Reply:
I don't see any use for equality checking of `Backoff` objects outside of reference equality, and calling `toString` on the class is going to log a lot of information. I am leaning towards just removing these annotations. If we really want `toString`, I can write one in the class.

Reply:
@lhotari - are you able to take a look at this again?